src/main/scopes/npm/find-installing-packages.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { DirectoryEnumerator } from '../../core/directory-enumerator.js'
+import { Logger } from '../../core/log/logger.js'
+import { runCommand } from '../../core/run-command.js'
+import { NoNodeModulesFoundError } from '../../exceptions/no-node-modules-found-error.js'
+import { findInstallersFromTree } from './find-installers-from-tree.js'
+import { hasNodeModulesFolder } from './has-node-modules-folder.js'
+import { InstallingPackage, PackageData } from './interfaces.js'
+
+const cache = new Map<string, Promise<InstallingPackage[]>>()
+
+/**
+ * Finds all packages within the project that installed the specified package at the specified
+ * version. This is done by finding the root-most project directory containing a node modules folder
+ * and running `npm ls` inside of it.
+ *
+ * If no installers are found after this project directory is searched, an empty array is returned.
+ *
+ * @param cwd - Current working directory. This must be inside of the root directory. This is an
+ * absolute path.
+ * @param root - Root-most directory. This is an absolute path.
+ * @param packageData - The name and exact version of the package to search for.
+ * @param logger - Logger instance.
+ * @returns A possibly empty array of installing packages.
+ */
+export async function findInstallingPackages(
```

Comment:
is there a difference between this file and src/main/core/find-installing-packages.ts?

Reply:
Oops, there should only be one of them. Just a merge bug. I'll remove the core one.